src/zabapgit_repo_impl.prog.abap

```diff
@@ -80,10 +82,20 @@ CLASS lcl_repo_online IMPLEMENTATION.
                         iv_total   = 1
                         iv_text    = 'Remote files' ) ##NO_TEXT.
 
-    lcl_git_porcelain=>pull( EXPORTING io_repo    = me
-                             IMPORTING et_files   = mt_remote
-                                       et_objects = mt_objects
-                                       ev_branch  = mv_branch ).
+    TRY.
+
+        lcl_git_porcelain=>pull( EXPORTING io_repo    = me
+                                 IMPORTING et_files   = mt_remote
+                                           et_objects = mt_objects
+                                           ev_branch  = mv_branch ).
+
+      CATCH lcx_exception INTO lx_exception.
+
+        delete_initial_online_repo( abap_true ).
```

Comment:
not really since this problem also occurs eg in lcl_services_abapgit->do_install to install abapgit/plugins which will than be displayed as installed at the tutorial page but aren`t